src/query/catalog/src/table.rs

```diff
@@ -218,6 +218,17 @@ pub trait Table: Sync + Send {
         )))
     }
 
+    fn build_prune_pipeline(
+        &self,
+        table_ctx: Arc<dyn TableContext>,
+        plan: &DataSourcePlan,
+        source_pipeline: &mut Pipeline,
+    ) -> Result<Option<Pipeline>> {
+        let (_, _, _) = (table_ctx, plan, source_pipeline);
```

Comment:
Maybe return error is better?

Reply:
I think it’s better to keep `Option` here because as we continue adding support for different engines, conditions like `table.get_table_info().meta.engine == *"FUSE" && ...` will inevitably grow. The disadvantage of using `Option` is that it doesn’t align with other operations. I’m open to changing it here if you think it’s important.